lib/orbit-common/cache.js

```diff
@@ -129,66 +130,74 @@ export default Class.extend({
   },
 
   /**
+   Returns whether a path has been removed from the document.
+
+   By default, this simply returns true if the path doesn't exist.
+   However, it may be overridden by an operations processor to provide more
+   advanced deletion tracking.
+
+   @method hasDeleted
+   @param path
+   @returns {Boolean}
+   */
+  hasDeleted: function(path) {
+    return !this.exists(path);
+  },
+
+  /**
    Transforms the document with an RFC 6902-compliant operation.
 
    Currently limited to `add`, `remove` and `replace` operations.
 
    @method transform
-   @param {Object} operation
-   @param {String} operation.op Must be "add", "remove", or "replace"
-   @param {Array or String} operation.path Path to target location
-   @param {Object} operation.value Value to set. Required for "add" and "replace"
-   @returns {Array or undefined} Array of inverse operations
+   @param {Array} [ops] Array of operations
+   @returns {TransformResult} The result of applying the operations.
    */
-  transform: function(operation) {
-    return this._transform(this._normalizeOperation(operation));
+  transform: function(ops) {
+    var result = new TransformResult();
+    this._applyOperations(normalizeOperations(ops), result);
```

Comment:
Maybe we could add `prepareOperations`, similar to the new `_prepareTransformOperations` on `Source`. We could call `normalizeOperations` first, then pass the ops into `prepareOperations`. That would allow for overriding.